crates/cast/src/cmd/mktx.rs

```diff
@@ -100,15 +126,50 @@ impl MakeTxArgs {
             .with_blob_data(blob_data)?;
 
         if raw_unsigned {
-            // Build unsigned raw tx
             let from = eth.wallet.from.ok_or_eyre("missing `--from` address")?;
-            let raw_tx = tx_builder.build_unsigned_raw(from).await?;
+            let raw_unsigned_hex = tx_builder.build_unsigned_raw(from).await?;
+
+            if let (Some(v), Some(r_hex), Some(s_hex)) = (v, r.as_ref(), s.as_ref()) {
+                let raw_unsigned_bytes = hex::decode(raw_unsigned_hex.trim_start_matches("0x"))?;
+                let mut buf = raw_unsigned_bytes.as_slice();
+                let unsigned_tx: UnsignedTransaction = UnsignedTransaction::decode(&mut buf)?;
+
+                let r_bytes = hex::decode(r_hex.trim_start_matches("0x"))?;
+                let s_bytes = hex::decode(s_hex.trim_start_matches("0x"))?;
+
+                if r_bytes.len() != 32 || s_bytes.len() != 32 {
+                    eyre::bail!("r and s must be 32 bytes each");
+                }
+
+                let r = U256::from_be_slice(&r_bytes);
+                let s = U256::from_be_slice(&s_bytes);
+                let v = U64::from(v);
+
+                let mut out = Vec::new();
+                let fields: &[&dyn alloy_rlp::Encodable] = &[
+                    &unsigned_tx.nonce,
+                    &unsigned_tx.gas_price,
+                    &unsigned_tx.gas_limit,
+                    &unsigned_tx.to,
+                    &unsigned_tx.value,
+                    &unsigned_tx.data,
+                    &v,
+                    &r,
+                    &s,
+                ];
+
+                alloy_rlp::encode_list::<&dyn alloy_rlp::Encodable, dyn alloy_rlp::Encodable>(
```

Comment:
this can removed entirely

this should be much simpler, just checking if the signature is there , decode it and then encode with signature:

https://github.com/alloy-rs/alloy/blob/ea2dc62620b34ce320e2232b2dbe7c218a1139ce/crates/consensus/src/transaction/rlp.rs#L53-L53